### PR TITLE
[Bromley] Add assisted collection event support.

### PR DESCRIPTION
**Pull request by dracos**
The Echo event bits of https://github.com/mysociety/fixmystreet/pull/3745

### CODE REVIEW

File: t/open311/endpoint/bromley.t [hunk 1]
```diff
@@ -1,10 +1,133 @@
...
 use warnings;
 
 BEGIN { $ENV{TEST_MODE} = 1; }
 
+use DateTime;
 use Test::More;
+use Test::MockModule;
+use JSON::MaybeXS;
+
+use constant EVENT_TYPE_ASSISTED => 2149;
```
Comment (davea) [inline t/open311/endpoint/bromley.t:34]: This doesn't seem to be used anywhere.

File: t/open311/endpoint/bromley.t [hunk 2]
```diff
@@ -1,10 +1,133 @@
...
+    # This is called when a test below makes a SOAP call, along with the data
+    # to be passed via SOAP to the server. We check the values here, then pass
+    # back a mocked result.
+    my ($cls, @args) = @_;
+    my $method = $args[0]->name;
+    if ($method eq 'PostEvent') {
+        my @params = ${$args[3]->value}->value;
+
+        my $client_ref = $params[1]->value;
+        like $client_ref, qr/^FMS-200012[34]$/;
```
Comment (davea) [inline t/open311/endpoint/bromley.t:47]: Does `FMS-2000124` ever get passed or can this be simplified?

**Review by davea — APPROVED** [review]
A few minor stylistic things, but 👍 overall.

File: t/open311/endpoint/bromley.t (hunk 1 above)
Comment (dracos) [reply t/open311/endpoint/bromley.t:34]: whoops, should have used it further down!

**Pull request merged**